### PR TITLE
Added computeSecondaryVariable to class Process

### PR DESCRIPTION
**Pull request by wenqing**
Follow up PR ufz/ogs#1468, this PR corrected the calculations of  heat flux and flow velocity
by
1 introducing a member of computeSecondaryVariable to class Process.
2. calling the new member after UncoupledProcessesTimeLoop

This is the 1st PR for the staggered scheme.

### CODE REVIEW

File: ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h [hunk 1]
```diff
@@ -130,40 +133,58 @@ class LiquidFlowLocalAssembler : public LiquidFlowLocalAssemblerInterface
...
+        static void calculateVelocity(
+            std::vector<std::vector<double>>& darcy_velocities,
+            Eigen::Map<const NodalVectorType> const& local_p,
+            ShapeMatrices const& sm, Eigen::MatrixXd const& perm,
+            unsigned const ip, double const mu, double const rho_g,
+            int const gravitational_axis_id);
     };
 
-    template <typename LaplacianAndGravityTermCalculator>
+    template <typename Calculator>
```
Comment (endJunction) [inline ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h]: It would be better if the name would be self-explanatory.

**Comment by endJunction** [comment]
Why can't you use the existing `pre/postTimestep` or `pre/postIteration` functions?

**Comment by wenqing** [comment]
@endJunction postIteration is called immediately after a successful nonlinear iteration loop. However, the calculation of velocity or heat flux is only needed for output on demand or for the coupled term. Alternatively, postIteration can be moved outside of the nonlinear solver, and the  computeSecondaryVariable can be merged to it.

File: ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h (hunk 1 above)
Comment (wenqing) [inline ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h]: How about ``LaplacianGravityVelocityCalculator``?

File: ProcessLib/UncoupledProcessesTimeLoop.cpp [hunk 2]
```diff
@@ -505,6 +505,8 @@ bool UncoupledProcessesTimeLoop::loop()
             nonlinear_solver_succeeded = solveOneTimeStepOneProcess(
                 x, timestep, t, delta_t, *spd, *_output);
 
+            pcs.computeSecondaryVariable(t, x);
```
Comment (wenqing) [inline ProcessLib/UncoupledProcessesTimeLoop.cpp]: A if-condition for coupling or output will be added in the next PR.

**Comment by endJunction** [comment]
@wenqing One change is pending.
Someone else please add a review.

File: ProcessLib/Process.h [hunk 3]
```diff
@@ -148,6 +151,9 @@ class Process
     virtual void preIterationConcreteProcess(const unsigned /*iter*/,
                                              GlobalVector const& /*x*/){}
 
+    virtual void computeSecondaryVariableConcrete(const double /*t*/,
+                                                  GlobalVector const& /*x*/) {};
```
Comment (norihiro-w) [inline ProcessLib/Process.h]: ; is not needed

File: ProcessLib/HeatConduction/HeatConductionProcess.cpp [hunk 4]
```diff
@@ -94,5 +94,14 @@ void HeatConductionProcess::assembleWithJacobianConcreteProcess(
         dx_dx, M, K, b, Jac);
 }
 
+void HeatConductionProcess::computeSecondaryVariableConcrete(const double t,
+                                                         GlobalVector const& x)
+{
+    DBUG("Compute the velocity for LiquidFlowProcess.");
```
Comment (norihiro-w) [inline ProcessLib/HeatConduction/HeatConductionProcess.cpp]: not LiquidFlowProcess

**Comment by norihiro-w** [comment]
:+1: besides some minor comments

File: ProcessLib/Process.h (hunk 3 above)
Comment (wenqing) [inline ProcessLib/Process.h]: removed.

File: ProcessLib/HeatConduction/HeatConductionProcess.cpp (hunk 4 above)
Comment (wenqing) [inline ProcessLib/HeatConduction/HeatConductionProcess.cpp]: Corrected.

**Comment by endJunction** [comment]
@wenqing Please fix the failing travis errors.

**Comment by wenqing** [comment]
jenkins test this please

**Comment by endJunction** [comment]
:fast_forward:

**Pull request merged**